docs/manual-EN/3.build-develop-and-administration/1.build/1.build-source-code.md

```diff
@@ -1,244 +1,200 @@
+$ git clone https://github.com/vesoft-inc/nebula.git
 ```
 
-- NEBULA\_BUILD\_VERSION: set the version number for the binary files
-
-For example, the following commands set the version number to `1.0.0`.
-
+If you don't care about the commit history of the repo, and to make the cloning faster, you could perform a shallow clone:
 ```bash
-bash> cd nebula/build
-bash> cmake -DNEBULA_BUILD_VERSION="1.0.0" ..
-bash> make -j$(nproc) && make install -j$(nproc)
+$ git clone --depth=1 https://github.com/vesoft-inc/nebula.git
 ```
 
-The following command checks the version number of the compiled binary files in `/usr/local/nebula/bin/` directory.
-
+### Configure and Build
 ```bash
-bash> /usr/local/nebula/bin/nebula --version
+$ cd nebula
+$ mkdir build
+$ cd build
+$ cmake -DENABLE_TESTING=OFF -DCMAKE_INSTALL_PREFIX=$PWD/install ..
+# N is suggested to the minimum of number of cores and half of available memory in Gigabytes
+$ make -jN
+$ make install
+$ ls install/
+etc/ bin/ share/ scripts/
 ```
 
-- ENABLE\_TESTING: specifies whether to compile the UT code.
-
-The default value is ON. If you don't want to compile, set it to OFF during cmake with the following command.
+Since C++ templates are heavily used by **Nebula Graph** and its third party dependencies, especially Folly, fbthrift and boost, the building is very time costly.
+For your reference, it is expected to take about 35 minutes in CPU time(less than 4 minutes with `-j16`), given an Intel E5-2697 v3 processor and unit tests are disabled.
 
-```bash
-bash> cd nebula/build
-bash> cmake -DENABLE_TESTING=OFF ..
-bash> make -j$(nproc) && make install -j$(nproc)
-```
+Until now, you might already have built **Nebula Graph** successfully. If so or not, we also provide ways to tweak the building process.
 
-- ENABLE\_ASAN: Add AddressSanitizer when compiling.
+## Ways to Tweak the Building
 
-The default value is OFF. You can set it to ON during cmake with the following command.
+### CMake Arguments/Variables
+We provide several options to make one tweak the building, while some of these are builtins from CMake. These arguments are used at the configure(cmake) stage, like `cmake -DArgument=Value ..`.
 
-```bash
-bash> cd nebula/build
-bash> cmake -DENABLE_ASAN=ON ..
-bash> make -j$(nproc) && make install -j$(nproc)
-```
+#### ENABLE_TESTING
+This option allows to enable or disable the build of unit tests. We suggest to turn it off if you just need the service modules of **Nebula Graph**.
+Optionals are `ON` and `OFF`, and `ON` by default.
 
-#### **Building Completed**
+#### ENABLE_ASAN
+This option enables or disables the ASan building, a.k.a AddressSanitizer, which is a memory error detector. It is meant to be used by **Nebula Graph** developers. This options is disabled by default.
 
-- If no errors are shown
+#### CMAKE_BUILD_TYPE
+There are a few building types supported:
+ * `Debug`, to build with debuginfo but without optimization, which is by default
```

Comment:
`Debug`, to build with debuginfo but without optimization by default

Reply:
_by default_ is to qualify _Debug_. I think it is more clear.